### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by billionsjoel**
- Create models
  - Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
  - Set up associations between models.
  - Use models to insert data
  - Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
  -    first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
  -    second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
- Create at least 4 posts written by one of the users you created by running the following code:
  -    first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
- Create at least 6 posts comments for one of the posts you created by running the following code:
-    Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
  - Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.
- Create custom methods
- Your models should include some additional methods.
- 
  - Users
  - A method that returns the 3 most recent posts for a given user.
  - Posts
  - A method that updates the posts counter for a user.
  - A method which returns the 5 most recent comments for a given post.
  - Comments
  - A method that updates the comments counter for a post.
  - Likes
  - A method that updates the likes counter for a post.
  - Go to rails c and check if your methods are working.
  - Additionally, you will need to make current user data available to all controllers.
  - 
  - Create a method called current_user in your ApplicationController.
  - A current_user method should return the first user from a database.
  - Don't worry about proper authentication at this time. You will add it in the upcoming projects.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,4 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :user, counter_cache: :posts_counter
-  has_many :likes, :comments
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_posts_count
+    author.increment!(:posts_counter)
+  end
```
Comment (Lameck1) [inline app/models/post.rb:11]: - [SUGGESTION] We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `update_posts_count` method every time a new post has been saved to the database, then make the `update_post_counter` method private.

  Something like this:
  
  ```ruby
  # app/models/post.rb
    after_save :update_posts_count
  
    private
  
    def update_posts_count
      author.increment!(:posts_counter)
    end
  ```

File: app/models/like.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Like < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  def update_post_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (Lameck1) [inline app/models/like.rb:10]: - [SUGGESTION] We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `update_post_likes_counter` method every time a new post has been saved to the database, then make the `update_post_likes_counter` method private.

  Something like this:
  
  ```ruby
  # app/models/like.rb
    after_save :update_post_likes_counter
  
    private
  
    def update_post_likes_counter
      post.increment!(:likes_counter)
    end
  ```

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
 class Comment < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  def update_post_comment_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (Lameck1) [inline app/models/comment.rb:10]: - [SUGGESTION] We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `update_post_comment_counter` method every time a new post has been saved to the database, then make the `update_post_comment_counter` method private.

  Something like this:
  
  ```ruby
  # app/models/comment.rb
    after_save :update_post_comment_counter
  
    private
  
    def update_post_comment_counter
      post.increment!(:comments_counter)
    end
  ```

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,4 +1,12 @@
...
+
+  def update_post_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (Lameck1) [inline app/models/comment.rb]: - [ ] REQUIRED CHANGE: Please delete this. It is a duplication of what is already being handled by the `like` model.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @billionsjoel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @billionsjoel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**